### ISSUE TITLE
SetTbarPostion Problems

### ISSUE THREAD

**Issue by gevorg-tsat** [obsproject/obs-websocket#1211]
### Operating System Info

Other

### Other OS

mac os 14.3.1

### OBS Studio Version

Other

### OBS Studio Version (Other)

30.0.2

### obs-websocket Version

5.1.0

### OBS Studio Log URL

https://obsproject.com/logs/qkqFUfDy0t993Iy2

### Expected Behavior

need some help c SetTBarPosition.
it doesnt work as i want, explain me please

so, 
when i send any position to obs, it works for microsecond and go back to its start position

then i noticed that SetTBarPosition works well when i make TBar "active". it's active when i move it anywhere and when Transition button looks inactive (as u can see on first screenshot)
but when Transition button is active(second screenshot), it again  works for microsecond and go back to its start position.
i dont want to move it manually all the time to make tbar "active".
<img width="182" alt="image" src="https://github.com/obsproject/obs-websocket/assets/72250109/5d3d01af-0670-4df2-bf33-40541fe1abf5">
<img width="175" alt="image" src="https://github.com/obsproject/obs-websocket/assets/72250109/3fc34f9b-453e-4b4f-8ddb-b413d4f103bf">

### Current Behavior

explained above

### Steps to Reproduce

my golang code sample:
goobs v1.2.3
```
_, err := obsClient.Transitions.SetTBarPosition(&transitions.SetTBarPositionParams{
        Position: &s.Position, // == 0.5
        Release:  &s.Release, //==true
    })
    if err != nil {
        logger.Info("Error occurred while setting TBar postion", zap.Error(err))
    }
```

**Comment by AVRC** [comment]
I have this exact issue and symptoms on Windows, OBS version 31.1.2(64 bit)
The command is accepted and returns code 100 but nothing happens unless I first move the T-Bar with my mouse off the left position.


11:54:37.970: [obs-websocket] [debug] [WebSocketServer::onMessage] Incoming message (decoded):
11:54:37.970: {
11:54:37.970:   "d": {
11:54:37.970:     "requestData": {
11:54:37.970:       "position": 0.45,
11:54:37.970:       "release": true
11:54:37.970:     },
11:54:37.970:     "requestId": "mzyqLKSU5u6oRCsa",
11:54:37.970:     "requestType": "SetTBarPosition"
11:54:37.970:   },
11:54:37.970:   "op": 6
11:54:37.970: }
11:54:37.971: [obs-websocket] [debug] [WebSocketServer::onMessage] Outgoing message:
11:54:37.971: {
11:54:37.971:   "d": {
11:54:37.971:     "requestId": "mzyqLKSU5u6oRCsa",
11:54:37.971:     "requestStatus": {
11:54:37.971:       "code": 100,
11:54:37.971:       "result": true
11:54:37.971:     },
11:54:37.971:     "requestType": "SetTBarPosition"
11:54:37.971:   },
11:54:37.971:   "op": 7
11:54:37.971: }

**Comment by Armandur** [comment]
Same problem here using OBS Studio 30.2.3 64bit on windows, nothing happens until I slide the t-bar a bit manually.

I was really hoping for this to work to implement at physical t-bar using an arduino.


`BUG:obsws_python.baseclient.ObsClient:Sending request {'op': 6, 'd': {'requestType': 'SetTBarPosition', 'requestId': 735, 'requestData': {'position': 0.8, 'release': True}}}
DEBUG:obsws_python.baseclient.ObsClient:Response received {'d': {'requestId': 735, 'requestStatus': {'code': 100, 'result': True}, 'requestType': 'SetTBarPosition'}, 'op': 7}
DEBUG:obsws_python.baseclient.ObsClient:Sending request {'op': 6, 'd': {'requestType': 'SetTBarPosition', 'requestId': 997, 'requestData': {'position': 0.9, 'release': True}}}
DEBUG:obsws_python.baseclient.ObsClient:Response received {'d': {'requestId': 997, 'requestStatus': {'code': 100, 'result': True}, 'requestType': 'SetTBarPosition'}, 'op': 7}
DEBUG:obsws_python.baseclient.ObsClient:Sending request {'op': 6, 'd': {'requestType': 'SetTBarPosition', 'requestId': 513, 'requestData': {'position': 1.0, 'release': True}}}
DEBUG:obsws_python.baseclient.ObsClient:Response received {'d': {'requestId': 513, 'requestStatus': {'code': 100, 'result': True}, 'requestType': 'SetTBarPosition'}, 'op': 7}
DEBUG:obsws_python.baseclient.ObsClient:Sending request {'op': 6, 'd': {'requestType': 'SetTBarPosition', 'requestId': 370, 'requestData': {'position': 1.0, 'release': True}}}
DEBUG:obsws_python.baseclient.ObsClient:Response received {'d': {'requestId': 370, 'requestStatus': {'code': 100, 'result': True}, 'requestType': 'SetTBarPosition'}, 'op': 7}
DEBUG:obsws_python.baseclient.ObsClient:Sending request {'op': 6, 'd': {'requestType': 'SetTBarPosition', 'requestId': 75, 'requestData': {'position': 0.9, 'release': True}}}
DEBUG:obsws_python.baseclient.ObsClient:Response received {'d': {'requestId': 75, 'requestStatus': {'code': 100, 'result': True}, 'requestType': 'SetTBarPosition'}, 'op': 7}
DEBUG:obsws_python.baseclient.ObsClient:Sending request {'op': 6, 'd': {'requestType': 'SetTBarPosition', 'requestId': 600, 'requestData': {'position': 0.8, 'release': True}}}
DEBUG:obsws_python.baseclient.ObsClient:Response received {'d': {'requestId': 600, 'requestStatus': {'code': 100, 'result': True}, 'requestType': 'SetTBarPosition'}, 'op': 7}
DEBUG:obsws_python.baseclient.ObsClient:Sending request {'op': 6, 'd': {'requestType': 'SetTBarPosition', 'requestId': 757, 'requestData': {'position': 0.7, 'release': True}}}
DEBUG:obsws_python.baseclient.ObsClient:Response received {'d': {'requestId': 757, 'requestStatus': {'code': 100, 'result': True}, 'requestType': 'SetTBarPosition'}, 'op': 7}
DEBUG:obsws_python.baseclient.ObsClient:Sending request {'op': 6, 'd': {'requestType': 'SetTBarPosition', 'requestId': 115, 'requestData': {'position': 0.6, 'release': True}}}
DEBUG:obsws_python.baseclient.ObsClient:Response received {'d': {'requestId': 115, 'requestStatus': {'code': 100, 'result': True}, 'requestType': 'SetTBarPosition'}, 'op': 7}
DEBUG:obsws_python.baseclient.ObsClient:Sending request {'op': 6, 'd': {'requestType': 'SetTBarPosition', 'requestId': 777, 'requestData': {'position': 0.5, 'release': True}}}
DEBUG:obsws_python.baseclient.ObsClient:Response received {'d': {'requestId': 777, 'requestStatus': {'code': 100, 'result': True}, 'requestType': 'SetTBarPosition'}, 'op': 7}
DEBUG:obsws_python.baseclient.ObsClient:Sending request {'op': 6, 'd': {'requestType': 'SetTBarPosition', 'requestId': 295, 'requestData': {'position': 0.4, 'release': True}}}
DEBUG:obsws_python.baseclient.ObsClient:Response received {'d': {'requestId': 295, 'requestStatus': {'code': 100, 'result': True}, 'requestType': 'SetTBarPosition'}, 'op': 7}
DEBUG:obsws_python.baseclient.ObsClient:Sending request {'op': 6, 'd': {'requestType': 'SetTBarPosition', 'requestId': 505, 'requestData': {'position': 0.3, 'release': True}}}
DEBUG:obsws_python.baseclient.ObsClient:Response received {'d': {'requestId': 505, 'requestStatus': {'code': 100, 'result': True}, 'requestType': 'SetTBarPosition'}, 'op': 7}
DEBUG:obsws_python.baseclient.ObsClient:Sending request {'op': 6, 'd': {'requestType': 'SetTBarPosition', 'requestId': 263, 'requestData': {'position': 0.2, 'release': True}}}
DEBUG:obsws_python.baseclient.ObsClient:Response received {'d': {'requestId': 263, 'requestStatus': {'code': 100, 'result': True}, 'requestType': 'SetTBarPosition'}, 'op': 7}
DEBUG:obsws_python.baseclient.ObsClient:Sending request {'op': 6, 'd': {'requestType': 'SetTBarPosition', 'requestId': 590, 'requestData': {'position': 0.1, 'release': True}}}
DEBUG:obsws_python.baseclient.ObsClient:Response received {'d': {'requestId': 590, 'requestStatus': {'code': 100, 'result': True}, 'requestType': 'SetTBarPosition'}, 'op': 7}
DEBUG:obsws_python.baseclient.ObsClient:Sending request {'op': 6, 'd': {'requestType': 'SetTBarPosition', 'requestId': 773, 'requestData': {'position': 0.0, 'release': True}}}
DEBUG:obsws_python.baseclient.ObsClient:Response received {'d': {'requestId': 773, 'requestStatus': {'code': 100, 'result': True}, 'requestType': 'SetTBarPosition'}, 'op': 7}
DEBUG:obsws_python.baseclient.ObsClient:Sending request {'op': 6, 'd': {'requestType': 'SetTBarPosition', 'requestId': 607, 'requestData': {'position': 0.0, 'release': True}}}
DEBUG:obsws_python.baseclient.ObsClient:Response received {'d': {'requestId': 607, 'requestStatus': {'code': 100, 'result': True}, 'requestType': 'SetTBarPosition'}, 'op': 7}
DEBUG:obsws_python.baseclient.ObsClient:Sending request {'op': 6, 'd': {'requestType': 'SetTBarPosition', 'requestId': 629, 'requestData': {'position': 0.1, 'release': True}}}
DEBUG:obsws_python.baseclient.ObsClient:Response received {'d': {'requestId': 629, 'requestStatus': {'code': 100, 'result': True}, 'requestType': 'SetTBarPosition'}, 'op': 7}
DEBUG:obsws_python.baseclient.ObsClient:Sending request {'op': 6, 'd': {'requestType': 'SetTBarPosition', 'requestId': 332, 'requestData': {'position': 0.2, 'release': True}}}
DEBUG:obsws_python.baseclient.ObsClient:Response received {'d': {'requestId': 332, 'requestStatus': {'code': 100, 'result': True}, 'requestType': 'SetTBarPosition'}, 'op': 7}
DEBUG:obsws_python.baseclient.ObsClient:Sending request {'op': 6, 'd': {'requestType': 'SetTBarPosition', 'requestId': 360, 'requestData': {'position': 0.3, 'release': True}}}
DEBUG:obsws_python.baseclient.ObsClient:Response received {'d': {'requestId': 360, 'requestStatus': {'code': 100, 'result': True}, 'requestType': 'SetTBarPosition'}, 'op': 7}
DEBUG:obsws_python.baseclient.ObsClient:Sending request {'op': 6, 'd': {'requestType': 'SetTBarPosition', 'requestId': 964, 'requestData': {'position': 0.4, 'release': True}}}
DEBUG:obsws_python.baseclient.ObsClient:Response received {'d': {'requestId': 964, 'requestStatus': {'code': 100, 'result': True}, 'requestType': 'SetTBarPosition'}, 'op': 7}
DEBUG:obsws_python.baseclient.ObsClient:Sending request {'op': 6, 'd': {'requestType': 'SetTBarPosition', 'requestId': 149, 'requestData': {'position': 0.5, 'release': True}}}
DEBUG:obsws_python.baseclient.ObsClient:Response received {'d': {'requestId': 149, 'requestStatus': {'code': 100, 'result': True}, 'requestType': 'SetTBarPosition'}, 'op': 7}
DEBUG:obsws_python.baseclient.ObsClient:Sending request {'op': 6, 'd': {'requestType': 'SetTBarPosition', 'requestId': 159, 'requestData': {'position': 0.6, 'release': True}}}
DEBUG:obsws_python.baseclient.ObsClient:Response received {'d': {'requestId': 159, 'requestStatus': {'code': 100, 'result': True}, 'requestType': 'SetTBarPosition'}, 'op': 7}
DEBUG:obsws_python.baseclient.ObsClient:Sending request {'op': 6, 'd': {'requestType': 'SetTBarPosition', 'requestId': 487, 'requestData': {'position': 0.7, 'release': True}}}
DEBUG:obsws_python.baseclient.ObsClient:Response received {'d': {'requestId': 487, 'requestStatus': {'code': 100, 'result': True}, 'requestType': 'SetTBarPosition'}, 'op': 7}
DEBUG:obsws_python.baseclient.ObsClient:Sending request {'op': 6, 'd': {'requestType': 'SetTBarPosition', 'requestId': 528, 'requestData': {'position': 0.8, 'release': True}}}
DEBUG:obsws_python.baseclient.ObsClient:Response received {'d': {'requestId': 528, 'requestStatus': {'code': 100, 'result': True}, 'requestType': 'SetTBarPosition'}, 'op': 7}
DEBUG:obsws_python.baseclient.ObsClient:Sending request {'op': 6, 'd': {'requestType': 'SetTBarPosition', 'requestId': 541, 'requestData': {'position': 0.9, 'release': True}}}
DEBUG:obsws_python.baseclient.ObsClient:Response received {'d': {'requestId': 541, 'requestStatus': {'code': 100, 'result': True}, 'requestType': 'SetTBarPosition'}, 'op': 7}
DEBUG:obsws_python.baseclient.ObsClient:Sending request {'op': 6, 'd': {'requestType': 'SetTBarPosition', 'requestId': 544, 'requestData': {'position': 1.0, 'release': True}}}
DEBUG:obsws_python.baseclient.ObsClient:Response received {'d': {'requestId': 544, 'requestStatus': {'code': 100, 'result': True}, 'requestType': 'SetTBarPosition'}, 'op': 7}
DEBUG:obsws_python.baseclient.ObsClient:Sending request {'op': 6, 'd': {'requestType': 'SetTBarPosition', 'requestId': 875, 'requestData': {'position': 1.0, 'release': True}}}
DEBUG:obsws_python.baseclient.ObsClient:Response received {'d': {'requestId': 875, 'requestStatus': {'code': 100, 'result': True}, 'requestType': 'SetTBarPosition'}, 'op': 7}
DEBUG:obsws_python.baseclient.ObsClient:Sending request {'op': 6, 'd': {'requestType': 'SetTBarPosition', 'requestId': 993, 'requestData': {'position': 0.9, 'release': True}}}
DEBUG:obsws_python.baseclient.ObsClient:Response received {'d': {'requestId': 993, 'requestStatus': {'code': 100, 'result': True}, 'requestType': 'SetTBarPosition'}, 'op': 7}`

**Comment by Pzx5073** [comment]
Same problem, I thought I was doing something wrong, moving the tbar a little from obs it starts to work

**Comment by Pzx5073** [comment]
Sorry for reopening this thread, I saw there was some progress and tried the new version of OBS 32.1.0 on Windows 10. With @djordje-mancic patch, the T-bar can be moved without problems. However, when you do a release: true (regardless of the position), the transition returns to the beginning. I don't know if anyone else has experienced this.

**Comment by h07d0q** [comment]
Hi,
just tested it on my end and it didn't solve my issue with trying to implement it in companion...
IIRC it still behaves like before:
- no change when I send a new position (release true or false)
- position changes only when I move the fader manually to any position in the UI, but the change is still not visible on the fader in the UI
Do I have to send a command first to start the transition?

**Comment by Pzx5073** [comment]
I don't know how it works in Companion; I assume you'll use a module. 
I do it using Node.js.

function mifunciontbar(slider) {

  const raw = Number(slider.value);
  const position = raw * 1023;
  const release = (position === 0 || position === 1023);
  const message = {
    op: 6,
    d: {
      requestType: "SetTBarPosition",
      requestId: "SetTBarPosition",
      requestData: {
        position: raw,
        release: release
      }
    }
  };
  sendCommand(JSON.stringify(message));
}
Basically, I send `release true` when I'm at the start or end point. During the movement, I send `false`.
With the new patch, I can now move the transition without moving the original OBS bar, but when I want to release it, it resets to 0.

**Comment by djordje-mancic** [comment]
@Pzx5073 The [pull request](https://github.com/obsproject/obs-studio/pull/13143) that fixes T-bar positioning on OBS' side hasn't been merged yet. Without it, changing the T-bar position via the websocket remains broken.
